drivers/spi/Kconfig.stm32

```diff
@@ -8,7 +8,7 @@
 
 menuconfig SPI_STM32
 	bool "STM32 MCU SPI controller driver"
-	depends on SPI && SOC_FAMILY_STM32
+	default y if SPI && SOC_FAMILY_STM32
```

Comment:
I understand this would be great, but this should go in a separate changeset

Reply:
Of course it can go to a different changeset. This PR includes various changes that I applied to make the drivers work with codegen (or better work). Currently it is a massive maintenance/ rebase effort that increases with every changeset to be managed. If codegen is accepted I can restructure the driver changes.